test/e2e/common/node/runtimeclass.go

```diff
@@ -56,7 +56,7 @@ var _ = SIGDescribe("RuntimeClass", func() {
 	*/
 	framework.ConformanceIt("should reject a Pod requesting a non-existent RuntimeClass", f.WithNodeConformance(), func(ctx context.Context) {
 		rcName := f.Namespace.Name + "-nonexistent"
-		expectPodRejection(ctx, f, e2eruntimeclass.NewRuntimeClassPod(rcName))
+		expectPodRejection(ctx, f, rcName)
```

Comment:
Unlike the delete case, this one should never pass.

You could change `expectPodRejection` to return an error, and then apply the appropriate expectation at each caller (i.e. this becomes somtehing like `gomega.Expect(checkPodRejection()).ToBe(Success())`, and an `Eventually(..)` call below